pandas/io/sql.py

```diff
@@ -692,37 +692,26 @@ def insert_data(self):
         column_names = list(map(str, temp.columns))
         ncols = len(column_names)
         data_list = [None] * ncols
-        blocks = temp._data.blocks
-
-        for b in blocks:
-            if b.is_datetime:
-                # return datetime.datetime objects
-                if b.is_datetimetz:
-                    # GH 9086: Ensure we return datetimes with timezone info
-                    # Need to return 2-D data; DatetimeIndex is 1D
-                    d = b.values.to_pydatetime()
-                    d = np.atleast_2d(d)
-                else:
-                    # convert to microsecond resolution for datetime.datetime
-                    d = b.values.astype("M8[us]").astype(object)
-            elif b.is_timedelta:
-                # numpy converts this to an object array of integers,
-                #  whereas b.astype(object).values would convert to
-                #  object array of Timedeltas
-                d = b.values.astype(object)
+
+        for i in range(len(temp.columns)):
+            ser = temp.iloc[:, i]
+            vals = ser._values
```

Comment:
sure, _ixs works here

Reply:
Although _ixs helps, the big win actually comes from avoiding creating a series if all you need are the values (as I did in the linked snippet)

Reply:
https://github.com/pandas-dev/pandas/pull/33252

Reply:
OK.  Until #33252 goes through, any objection to _ixs here?

Reply:
Yes, whichever gets merged second can be update to use it here

Reply:
I really don't like using private methods like this, what is wrong with .iloc here?

Reply:
Using methods *internally* in frame.py is one thing; however this should be using a public API (and likely this makes almost no difference here).

Reply:
i agree with jeff on this one

Reply:
It's exactly the goal of the helper function I am adding in https://github.com/pandas-dev/pandas/pull/33252 to be used in situations where only the array values are needed, like here is the case.

So if we are adding such helper method, why not use it? (the reason I did the PR is because I want to see it used in several places, and many of those places will be *outside* of frame.py, eg in the ops code)

If the privateness of the method is a problem, let's discuss making it public. Although I personally think that is not needed right now (and we are using plenty of semi-private methods on DataFrame/Series outside of `frame.py` already)

Reply:
lets revisit that once #33252 is merged

Reply:
This PR certainly does not need to wait on #33252 being merged, but I still first want to have the discussion whether we find this an appropriate place to use that function here, once it exists.

Reply:
I don't think we should be using internal functions *except* in the internals. This is not internal by any stretch (would say that pandas.io is offlimits entirely)